src/Promise/FulfilledPromise.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace Http\Client\Utils\Promise;
+
+use Http\Client\Exception;
+use Http\Client\Promise;
+use Psr\Http\Message\ResponseInterface;
+
+/**
+ * A promise already fulfilled
+ *
+ * @author Joel Wurtz <jwurtz@jolicode.com>
+ */
+class FulfilledPromise implements Promise
```

Comment:
Totally adapter specific i a default implementation is really hard to provide specially for the wait method